source/common/quic/active_quic_listener.cc

```diff
@@ -65,6 +65,17 @@ ActiveQuicListener::ActiveQuicListener(
                                    quic::QuicCryptoServerConfig::ConfigOptions());
   auto alarm_factory =
       std::make_unique<EnvoyQuicAlarmFactory>(dispatcher_, *connection_helper->GetClock());
+  // Set the socket to report incoming ECN.
```

Comment:
Are we thinking that we're going to do this by default, or should ECN support be a config option? I suspect we might want the latter. If so, we'd want to add a new option to `QuicProtocolOptions` in `api/envoy/config/listener/v3/quic_config.proto`. We can read the value of this field and pass it into the `ActiveQuicListener` constructor in ActiveQuicListenerFactory. See `packets_to_read_to_connection_count_ratio` as an example.

@alyssawilk What's your take on this?

Reply:
I am of the opinion that we should do this 100% of the time if the code path normally uses recvmsg, and possibly even if it doesn't, but I'll defer to those who understand Envoy deployments better.

Reply:
I guess I could use a bit more detail on the pros and cons.  I assume it's something like "better congestion control  vs a teeny tiny bit of CPU"?
I'd say we should definitely either config guard or runtime guard but until I know the above I can't guess if we need a permanent knob or not.

Reply:
Alright, I added the protobuf for now

Reply:
Can you address @alyssawilk 's question:
> I guess I could use a bit more detail on the pros and cons. I assume it's something like "better congestion control vs a teeny tiny bit of CPU"?

Reply:
Yes, that is the anticipated tradeoff. I added the config option so I guess the question is moot.

Reply:
Possibly. But config knobs are "forever", so we should be sure about them. If it's only a tiny bit of CPU, then maybe we should just enable it behind a runtime guard, which is temporary. Both are plausible. Let's let Alyssa chime in.

Reply:
OK, switched from protobuf to runtime guard